From: Jay Vosburgh <fubar@us.ibm.com>
Date: Fri, 31 May 2013 11:57:26 +0000
Subject: net/core: __hw_addr_create_ex does not initialize sync_cnt
Patch-mainline: v3.10-rc5
Git-commit: 9747ba6636be8a7e8ba83a1fb231d061ca318e4f
References: bsc#971433

The sync_cnt field is not being initialized, which can result
in arbitrary values in the field.  Fixed by initializing it to zero.

Signed-off-by: Jay Vosburgh <fubar@us.ibm.com>
Reviewed-by: Vlad Yasevich <vyasevic@redhat.com>
Tested-by: Shawn Bohrer <sbohrer@rgmadvisors.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Michal Kubecek <mkubecek@suse.cz>

---
 net/core/dev_addr_lists.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/net/core/dev_addr_lists.c b/net/core/dev_addr_lists.c
index af34264dffd0..c7344e0b81df 100644
--- a/net/core/dev_addr_lists.c
+++ b/net/core/dev_addr_lists.c
@@ -39,6 +39,7 @@ static int __hw_addr_create_ex(struct netdev_hw_addr_list *list,
 	ha->refcount = 1;
 	ha->global_use = global;
 	ha->synced = sync;
+	ha->sync_cnt = 0;
 	list_add_tail_rcu(&ha->list, &list->list);
 	list->count++;
 
-- 
2.8.1

